File: fun/dice_odd_even_sim/odd_when_even.py
#!/usr/bin/env python3
# For a dice with 6 equiprobable outcomes, what is the probability of odd when even
# we model Odd = False, Even = True
from random import choice


def roll() -> bool:
    return choice([True, False])


def experiment():
    even = True
    odd = False

    total = 100000
    wins = 0

    for _ in range(total):
        a = roll()
        b = roll()
        if a == even and b == odd:
            wins += 1

    print(f"{wins}/{total} => {wins/total}")


if __name__ == '__main__':
    for _ in range(10):
        experiment()

File: fun/dice_odd_even_sim/odd_when_even.clj
#!/usr/bin/env clj
;; For a dice with 6 equiprobable outcomes, what is the probability of odd when even
;; we model Odd = 1, Even = 2

(defn roll [] (rand-nth [1 2]))

(defn win? [a b]
   (and (even? a) (odd? b)))


(defn run []
    (let [a (roll)
          b (roll)]
          (win? a b)))


(defn experiment [total]
    (->>
       run
       repeatedly
       (take total)
       (filter true?)
       count))


(doseq [_ (range 0 10)]
  (let [total 10000
      n (experiment total)]

      (println n "/" total " => " (double (/ n total)))))
